jwql/website/apps/jwql/monitor_pages/monitor_bias_bokeh.py

```diff
@@ -93,35 +97,24 @@ def retrieve_trending_data(self, aperture):
         """
         # Query database for all data in bias stats with a matching aperture,
         # and sort the data by exposure start time.
-        tmp_trending_data = session.query(self.stats_table.amp1_even_med,
-                                          self.stats_table.amp1_odd_med,
-                                          self.stats_table.amp2_even_med,
-                                          self.stats_table.amp2_odd_med,
-                                          self.stats_table.amp3_even_med,
-                                          self.stats_table.amp3_odd_med,
-                                          self.stats_table.amp4_even_med,
-                                          self.stats_table.amp4_odd_med,
-                                          self.stats_table.expstart,
-                                          self.stats_table.uncal_filename) \
-            .filter(self.stats_table.aperture == aperture) \
-            .order_by(self.stats_table.expstart) \
-            .all()
-
-        session.close()
+        columns = ['amp1_even_med', 'amp1_odd_med', 'amp2_even_med', 'amp2_odd_med',
+                   'amp3_even_med', 'amp3_odd_med', 'amp4_even_med', 'amp4_odd_med',
+                   'expstart', 'uncal_filename']
+        tmp_trending_data = self.stats_table.objects.filter(aperture__iexact=aperture).order_by('expstart').all().values(*columns)
 
         # Convert the query results to a pandas dataframe
-        self.trending_data = pd.DataFrame(tmp_trending_data, columns=['amp1_even_med', 'amp1_odd_med',
-                                                                      'amp2_even_med', 'amp2_odd_med',
-                                                                      'amp3_even_med', 'amp3_odd_med',
-                                                                      'amp4_even_med', 'amp4_odd_med',
-                                                                      'expstart_str', 'uncal_filename'])
-        uncal_basename = [os.path.basename(e) for e in self.trending_data['uncal_filename']]
-        self.trending_data['uncal_filename'] = uncal_basename
-
-        # Add a column of expstart values that are datetime objects
-        format_data = "%Y-%m-%dT%H:%M:%S.%f"
-        datetimes = [datetime.strptime(entry, format_data) for entry in self.trending_data['expstart_str']]
-        self.trending_data['expstart'] = datetimes
+        if len(tmp_trending_data) != 0:
+            self.trending_data = pd.DataFrame.from_records(tmp_trending_data)
+            uncal_basename = [os.path.basename(e) for e in self.trending_data['uncal_filename']]
+            self.trending_data['uncal_filename'] = uncal_basename
+
+            # Add a column of expstart values that are datetime objects
+            format_data = "%Y-%m-%dT%H:%M:%S.%f"
```

Comment:
If there are complaints in the log about missing timezone info, you could update the strings and use the %z option in strptime. For this to work, the strings would have to change to be e.g. '2021-09-01 15:27:05.004573 +0000'

Reply:
No complaints in the log about timezones. expstart entries are strings rather than datetime objects.